params/superchain.go

```diff
@@ -75,46 +75,32 @@ func LoadOPStackChainConfig(chainID uint64) (*ChainConfig, error) {
 		TerminalTotalDifficultyPassed: true,
 		Ethash:                        nil,
 		Clique:                        nil,
-		Optimism: &OptimismConfig{
-			EIP1559Elasticity:        6,
-			EIP1559Denominator:       50,
-			EIP1559DenominatorCanyon: newUint64(250),
-		},
+		Optimism:                      nil, // Set below
+	}
+
+	// Optimism parameters are loaded from the genesis
+	gen, err := superchain.LoadGenesis(chainID)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load genesis definition for chain %d: %w", chainID, err)
 	}
+	out.Optimism = (*OptimismConfig)(gen.Config.Optimism)
 
 	// special overrides for OP-Stack chains with pre-Regolith upgrade history
 	switch chainID {
-	case OPGoerliChainID:
-		out.LondonBlock = big.NewInt(4061224)
-		out.ArrowGlacierBlock = big.NewInt(4061224)
-		out.GrayGlacierBlock = big.NewInt(4061224)
-		out.MergeNetsplitBlock = big.NewInt(4061224)
-		out.BedrockBlock = big.NewInt(4061224)
-		out.RegolithTime = &OptimismGoerliRegolithTime
-		out.Optimism.EIP1559Elasticity = 10
 	case OPMainnetChainID:
 		out.BerlinBlock = big.NewInt(3950000)
 		out.LondonBlock = big.NewInt(105235063)
 		out.ArrowGlacierBlock = big.NewInt(105235063)
 		out.GrayGlacierBlock = big.NewInt(105235063)
 		out.MergeNetsplitBlock = big.NewInt(105235063)
 		out.BedrockBlock = big.NewInt(105235063)
-	case BaseGoerliChainID:
-		out.RegolithTime = &BaseGoerliRegolithTime
-		out.Optimism.EIP1559Elasticity = 10
-	case baseSepoliaChainID:
-		out.Optimism.EIP1559Elasticity = 10
-	case baseGoerliDevnetChainID:
-		out.RegolithTime = &baseGoerliDevnetRegolithTime
-	case pgnSepoliaChainID:
-		out.Optimism.EIP1559Elasticity = 2
-		out.Optimism.EIP1559Denominator = 8
+		out.Optimism = &OptimismConfig{
+			EIP1559Elasticity:        6,
+			EIP1559Denominator:       50,
+			EIP1559DenominatorCanyon: newUint64(250),
+		}
```

Comment:
Why does this still need to be set explicitly for OPM?

Reply:
Good question. I think this may not be necessary after all, removed it in [2d89785](https://github.com/ethereum-optimism/op-geth/pull/371/commits/2d89785aecb66432feac1e1016d4753847a67abb).

Reply:
That is possible, and that is in fact the current state of things at the tip of my PR branches. 

It's not a big lift to go either way, I just wonder about leaving the registry data in an "honest" state. In the current state, OPM has a custom data schema which doesn't match up to any existing code path in the `add-chain/compress-genesis` tool.

Reply:
So that sounds like fixing OPM genesis data is the cleaner approach? I'm also happy to do it in a follow up, to get this PR in with the manual OPM override.

Reply:
In the end, going with the new approach of storing the config separately from the genesis file, this problem went away.